x/ratelimit/README.md

```diff
@@ -125,6 +129,11 @@ For a more detailed explanation, see the[ ICS-20 ADR](https://github.com/cosmos/
     - (1) Remove Prefix: `transfer/channel-Z/ujuno`
     - (2) Hash: `ibc/...`
 
+## Packet Failures and Timeouts
+When a transfer is sent, the `Outflow` for the corresponding rate limit is incremented. Consequently, if the transfer fails on the host or times out, the change in `Outflow` must be reverted. However, the decrement is only necessary if the acknowledgement or timeout is returned in the same quota window that the packet was originally sent from.
+
+To keep track of whether the packet was sent in the same quota, the sequence number of all pending packets are stored. This is implemented by recording the sequence number of a SendPacket as it is sent, and then removing that list of sequence numbers each time the rate limit is reset at the end of the quota. Additionally, the sequence numbers are also removed when after an acknowledgement or timeout (a step that is not entirely necessary, but does reduce the size of the state).
```

Comment:
Since we use the channel-ID on stride, I think channel ID + sequence number is unique. I don't think it's possible to have two different ports on the same channel-id
My mental model is that a port is really just an alias for channel

Reply:
Reading the IBC code, it seems the sequence number is unique for a channel/port pair, not a channel. I think we assume that only channel-id/transfer packets are sent across the channel, which seems safe because there's no code path that can send channel-id/other-port packets across the channel (and if that's true, then sequence number is unique per channel).

However, I think sending packets across the channel from another port is possible (we should verify this). Channel/ports definitely don't map 1:1, for example you can have two transfer channels: channel-1 and channel-2 on connection-1.

If it is possible for packets from other ports to be sent across the channel, it seems like using the full uniqueness key is a bit more defensive

Reply:
Actually this seems wrong
```
interface ChannelEnd {
  state: ChannelState
  ordering: ChannelOrder
  counterpartyPortIdentifier: Identifier
  counterpartyChannelIdentifier: Identifier
  connectionHops: [Identifier]
  version: string
}
```
A `ChannelEnd` has a `counterpartyPortIdentifier`, so it seems channels do map onto a single port on each chain?

Reply:
https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/README.md

Reply:
chatted offline, in agreement that channel's do map to a single port so channel ID is sufficient as a unique identifier